odoo/cli/i18n.py

```
@@ -0,0 +1,164 @@
+        db_name = db_names[0]
+
+        subcommand_handler = {
+            'import': self._import,
+            'export': self._export,
+            'loadlang': self._loadlang,
+        }[parsed_args.subcommand]
+
+        # Run subcommand
+        registry = Registry.new(db_name)
+        with closing(registry.cursor()) as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            subcommand_handler(env, parsed_args)
+
+    def _import(self, env, parsed_args):
+        if parsed_args.format in ('tgz', 'csv') and not parsed_args.lang:
+            Command.die("Language must be specified when '*.tgz' and '*.csv' files are imported.")
+        if parsed_args.format == 'csv':
+            # The default limit for CSV fields in the module is 128KiB, which is not
+            # quite sufficient to import images to store in attachment. 500MiB is a
+            # bit overkill, but better safe than sorry
+            csv.field_size_limit(500 << 20)
+
+        translation_importer = odoo.tools.translate.TranslationImporter(env.cr)
+        for filename in parsed_args.files:
+            if parsed_args.format == 'pot':
+                lang = parsed_args.lang
+            elif Path(filename).suffixes[-1:] == ['.po']:
+                code = Path(filename).stem
+                domain = ['|', ('code', '=', code), ('code', 'ilike', f'{code}%')]
+                lang = env['res.lang'].search(domain, limit=1).code
+            translation_importer.load_file(filename, lang)
+        translation_importer.save(overwrite=not parsed_args.only_new)
+
+    def _export(self, env, parsed_args, logging_enabled=True):
+
+        # When outputting to stdout, disable the logging
+        if parsed_args.to_stdout and logging_enabled:
+            logging.disable(logging.ERROR)
+            try:
+                self._export(env, parsed_args, logging_enabled=False)
+            finally:
+                logging.disable(logging.NOTSET)
+            return
```

if I am not mistaken, we write the logs on stderr, so there shouldn't be obligated to disable the logging, if the user wants to suppress the errors he can redirect 2 to /dev/null and achieve the same

---

That the method recalls itself with other arguments is kinda a gotcha. It took bit 5 seconds too much to understand what was going on.

Suggestion:

```py
def _export(...):
    if logging_enabled:
        _export_stdout(...)
    else:
        _export_file(...)

def _export_stdout(...):
    ...
    _export_file(...)
    ...

def _export_file(...):
    ...
```

---

But maybe that's not necessary if we don't suppress the logs.

---

Otherwise, if we really want to suppress the logs:

```py
def _export(...):
    with contextlib.ExitStack() as stack:
        if not logging_enabled:
            stack.enter(mute_logger(''))
        ...
```